eppo/src/main/java/cloud/eppo/android/ConfigurationStore.java

```diff
@@ -57,35 +60,28 @@ public boolean loadFromCache(InitializationCallback callback) {
                     RandomizationConfigResponse configResponse = gson.fromJson(reader, RandomizationConfigResponse.class);
                     reader.close();
                     if (configResponse == null || configResponse.getFlags() == null) {
-                        // Invalid cached configuration, initialize as an empty map and delete file
                         throw new JsonSyntaxException("Configuration file missing flags");
                     }
                     flags = configResponse.getFlags();
                     updateConfigsInSharedPrefs();
                 }
                 Log.d(TAG, "Cache loaded successfully");
+                callback.onCacheLoadSuccess();
             } catch (Exception e) {
                 Log.e(TAG, "Error loading from local cache", e);
                 cacheFile.delete();
-
-                if (callback != null) {
-                    callback.onError("Unable to load config from cache");
-                }
-            }
-
-            if (callback != null) {
-                callback.onCompleted();
+                callback.onCacheLoadFail();
             }
         });
-        return true;
     }
 
     public void setFlags(Reader response) {
         RandomizationConfigResponse config = gson.fromJson(response, RandomizationConfigResponse.class);
-        flags = config.getFlags();
-        if (flags == null) {
+        if (config == null || config.getFlags() == null) {
```

Comment:
Re-arrange to handle the null config edge case (shouldn't happen, but being defensive)